pkcs11/src/config/logging.rs

```diff
@@ -1,37 +1,166 @@
-use super::config_file::P11Config;
+use log::{info, warn};
+use syslog::{BasicLogger, Formatter3164};
 
-// output to stdout and a file
-pub fn configure_logger(config: &P11Config) {
-    let mut builder = env_logger::Builder::from_default_env();
+use super::{
+    config_file::{LogLevel, P11Config},
+    initialization::InitializationError,
+};
 
-    // set the log level
+// output to stdout, a file or syslog
+pub fn configure_logger(config: &Result<P11Config, InitializationError>) {
+    let Ok(config) = config else {
+        // On error, first try logging to syslog
+        if syslog::init_unix(syslog::Facility::LOG_USER, log::LevelFilter::Info).is_ok() {
+            return;
+        };
+        // Otherwise try to log to stderr
+        env_logger::try_init().ok();
+        return;
```

Comment:
Maybe it would make sense to log both to stderr and syslog.  In some cases, messages on stderr are more intutive.  Do you know if `pkcs11-tool` passes through `stderr` messages per default?

Reply:
It does pass through `stderr`. Ok, i'll adapt it to log to both by default.